### PR TITLE
test: enrich and unbrittle Persistence tests of State.Core

### PR DESCRIPTION
**Pull request by pdobacz**
Instead of doing a lot of `db_updates` assertions lets run them through the persistence layer (using public APIs) and see if it works as intended.

Reduces the "reaching-into-the-guts" feeling with all the `{:put, :utxo...` etc assertions.

Now persistence per-se is tested in `State.PersistenceTest` and what is left in `State.CoreTest` is more oriented towards various details of the behavior (like flushing the mempool etc.).

Also - battle testing the approach before attacking `ExitProcessor.CoreTest` in a similar manner

### CODE REVIEW

File: apps/omg_db/test/support/db_case.ex [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  @moduledoc """
+  Defines the useful common setup for all `...PersistenceTests`:
+   - starts temporary file handler `:briefly`
+   - creates temp dir with that
+   - initializes the low-level LevelDB storage and starts the test DB server
+  """
+
+  use ExUnit.CaseTemplate
+
+  using do
```
Comment (pnowosie) [inline apps/omg_db/test/support/db_case.ex]: Wow, this's interestning! So by `use` this file in tests we have `setup_all` and `setup` callbacks fired by ExUnit :question: 
Nice!

File: apps/omg_api/test/state/persistence_test.exs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    {state, db_updates}
+  end
+
+  defp persist_deposit(state, deposits, db_pid) do
+    {state, db_updates} = deposit(state, deposits)
+    assert :ok = OMG.DB.multi_update(db_updates, db_pid)
+    state
+  end
+
+  defp form(state) do
```
[the remark this answers is not in the document]
Comment (pdobacz) [reply apps/omg_api/test/state/persistence_test.exs]: i wanted to be super-concise here, hence I'm not 100% in line with the original API. These are private functions which drive this test only and are taylor made for this test.

Would that make this convention acceptable? I'm thinking of a DSL-like crafting of the tests, I think also [this](https://github.com/omisego/elixir-omg/blob/master/apps/omg_api/test/ethereum_event_listener/core_test.exs) takes this approach, which I'm inclined towards.

File: apps/omg_api/test/state/persistence_test.exs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    {:ok, {_, _, db_updates}, state} = Core.form_block(@interval, state)
+    {state, db_updates}
+  end
+
+  defp persist_form(state, db_pid) do
```
[the remark this answers is not in the document]
Comment (pdobacz) [reply apps/omg_api/test/state/persistence_test.exs]: see other thread

File: apps/omg_db/test/support/db_case.ex (hunk 1 above)
Comment (pdobacz) [reply apps/omg_db/test/support/db_case.ex]: :thinking: it's possible that this empty using isn't necessary at all, I might have left it here by mistake

Comment (pdobacz) [reply apps/omg_db/test/support/db_case.ex]: oh, and on `ExUnit.CaseTemplate` - yes - this is the more "standard" way of doing what we're doing with `ex_unit_fixtures`, I wanted to try it out with the intention of preferring `ExUnit.CaseTemplate` wherever it's enough (`deffixture` offers a lot more in terms of composability of the steup routines)

**Pull request merged**